### DISCUSSION TITLE
How to correctly handle InvalidScope in a file required by a class?

### DISCUSSION THREAD

**Discussion by kkmuffme** [vimeo/psalm#8401] [General]
```php
class foo {
    protected function display_something() {
        echo "hello";
    }
    
    public function output_template() {
        require 'my-template.php';
    }
}
```

"my-template.php" uses `$this->display_something()` and I get InvalidScope error.
What is the best way to document that this file is loaded inside the "foo" class?

**Comment by orklah** [comment]
You should be able to use the undocumented `/** @psalm-scope-this foo */`

IIRC, it tells Psalm you're inside that class already

**Reply by kkmuffme** [reply]
Thanks, will try that
